src/shared/parameter_v2/parameter.bzl

```diff
@@ -0,0 +1,43 @@
+def _generate_dynamic_parameters_impl(ctx):
+    output_file = ctx.outputs.generated_parameter_file
+
+    args = ctx.actions.args()
+    args.add("--output_file", output_file)
+
+    if output_file.extension == "hpp":
+        args.add("-cpp")
```

Comment:
We're going to use .hpp files now? Seems strange since we've been using .h for cpp for ages

Reply:
I agree, this seems odd.

Reply:
I  will split this rule up as Jon suggested above so the file extension won't matter.

Reply:
resolved by splitting into separate rules (macros in this case) for c and cpp. See above comment.